python/ray/tune/utils/placement_groups.py

```diff
@@ -104,10 +119,19 @@ class PlacementGroupFactory:
     """
 
     def __init__(self,
-                 bundles: List[Dict[str, float]],
+                 bundles: List[Dict[str, Union[int, float]]],
                  strategy: str = "PACK",
                  *args,
                  **kwargs):
+        assert len(bundles) > 0, (
+            "Cannot initialize a PlacementGroupFactory with zero bundles.")
+        if not bundles[0]:
+            # This is when trainable itself doesn't need resources.
+            self._head_bundle_is_empty = True
+            bundles.pop(0)
```

Comment:
If you pop the first bundle, isn't it causing problem when you schedule actors on bundle index 0?

Reply:
You could be right now that you explained to me what that bundle index 0 argument means.
Although I don't observe any issues when tests run through this branch; we should clean up the API semantics here. i.e. Applications should be able to specify the case where actor requires 0 resources.